### ISSUE TITLE
USER ISSUE: Removing a block, then immediately right-clicking a stockpile also takes out items of the stockpile

### ISSUE THREAD

**Issue by Yogu** [StrangeLoopGames/EcoIssues#24470]
\
**User Id: (Not logged user)**
**Version:** 0.10.0.3 beta release-547

**Steps to Reproduce:**
1. Place a block next to a stockpile
2. Left-click that block with a hammer to remove it
3. Immediately, right-click the stockpile to put the block into it

**Expected behavior:**
The block should be put into the stockpile, and nothing else should happen

**Actual behavior:**
The block is put into the stockpile, but then the hand fills with items out of the inventory, as if you would have left-clicked the stockpile

**Do you have mods installed? Does the issue happen when no mods are installed?:**
No mods

**Please provide log files. Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Getting-the-Eco-game-log-file (probably does not need log files)

This happened to me a few times until I realized what's going on. It's quite easy to trigger this bug when cleaning up a structure, and it does not look intentional (you basically get two left-click actions by only clicking once).

It happens whether or not the block removal was successful or not (i.e. it also occurs if you were not able to remove the block because you carry different items). I think the behavior is a bug in both cases.

**Comment by D3nnis3n** [comment]
Do you have an SLG account? You made some very good reports lately (even though I unfortunately had to close some as either working as intended or wontfix) and if you have I could give you a little goodie. :)

**Comment by Yogu** [comment]
Yep, made a SLG account recently, id is slg238555. Glad you appreciate the reports :) I really like how open everything is in this project.
